content/en/docs/reference/using-api/api-concepts.md

```diff
@@ -937,15 +938,74 @@ rules:
 
 See [Authorization Overview](/docs/reference/access-authn-authz/authorization/).
 
-## Server Side Apply
+## Updates to existing resources {#patch-and-apply}
+
+You can overwrite (**update**) an existing resource - for example, a ConfigMap -
+using an HTTP PUT. For a PUT request, it is the client's responsibility to specify
+the `resourceVersion` (taking this from the object being updated). Kubernetes uses
+that `resourceVersion` information so that the API server can detect lost updates
+and reject requests made by a client that is out of date with the cluster.
+In the event that the resource has changed (the `resourceVersion` the client
+provided is stale), the API server returns a `409 Conflict` error response.
+
+Instead of sending a PUT request, the client can send an instruction to the API
```

Comment:
Unfortunately, we're not really addressing the question of "Why would I do one instead of the other?"

Reply:
Making this a little longer than I would want but I'm being asked these questions ALL THE TIME, and I'm still looking for the perfect reference to point to, so let's take that chance:

UPDATE is great and easy, a couple of problems:
1. You might accidentally drop fields if you decode into Go objects that are not in the latest version (new fields are going to be dropped)
2. If there's a lot of contention on the object (even on a part that you're not trying to edit), you might have trouble sending the update. (you are sort of mentioning that in the next sentence but I don't know if that's super clear).
3. Related to 2, the object might be a lot bigger, which when there is contention, makes it even harder to update quickly.
4. You can easily depend on some values of the object

PATCH is great because:
1. You're sending only the piece that you want to change, the change is smaller
2. You don't depend on the rest of the object (as you mentioned in the next sentence)
3. Might require you to manually set the resourceVersion if you want to be super careful
5. you need to be careful when building the patch to not drop un-deserialized fields (order of the operation matters)
6. you need more local logic to build the patch (depending on language and presence of proper json patch libraries)
7. You can still fairly well depend on other values of the object
8. If you're using json-patch, you can use `test` conditions to careful craft specific update conditions

Apply is awesome because:
1. Doesn't require a GET before-hand
2. You still get conflicts if you change something that is unexpected, which significantly reduces the retry-on-error loop for conflicts, but they can easily be forced
3. Extremely easy to build
4. Does not work at all for values that depend on a current value of the object

Reply:
I added a new section, see fragment `#update-mechanism-choose` of the revised doc.